connectivity/factory/factory.go

```diff
@@ -0,0 +1,248 @@
+
+	//go:embed manifests/client-egress-to-cidr-external-deny.yaml
+	clientEgressToCIDRExternalDenyPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http.yaml
+	clientEgressL7HTTPPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http-named-port.yaml
+	clientEgressL7HTTPNamedPortPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-tls.yaml
+	clientEgressL7TLSPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http-matchheader-secret.yaml
+	clientEgressL7HTTPMatchheaderSecretYAML string
+
+	//go:embed manifests/echo-ingress-from-cidr.yaml
+	echoIngressFromCIDRYAML string
+)
+
+var (
+	clientLabel  = map[string]string{"name": "client"}
+	client2Label = map[string]string{"name": "client2"}
+
+	// Test order is important
+	primaryTests = []factory{
+		noUnexpectedPacketDrops{},
+		noPolicies{},
+		noPoliciesFromOutside{},
+		noPoliciesExtra{},
+		allowAllExceptWorld{},
+		clientIngress{},
+		clientIngressKnp{},
+		allowAllWithMetricsCheck{},
+		allIngressDeny{},
+		allIngressDenyFromOutside{},
+		allIngressDenyKnp{},
+		allEgressDeny{},
+		allEgressDenyKnp{},
+		allEntitiesDeny{},
+		clusterEntity{},
+		clusterEntityMultiCluster{},
+		hostEntityEgress{},
+		hostEntityIngress{},
+		echoIngress{},
+		echoIngressFromOutside{},
+		echoIngressKnp{},
+		clientIngressIcmp{},
+		clientEgress{},
+		clientEgressKnp{},
+		clientEgressExpression{},
+		clientEgressExpressionKnp{},
+		clientWithServiceAccountEgressToEcho{},
+		clientEgressToEchoServiceAccount{},
+		toEntitiesWorld{},
+		toCidrExternal{},
+		toCidrExternalKnp{},
+		fromCidrHostNetns{},
+		echoIngressFromOtherClientDeny{},
+		clientIngressFromOtherClientIcmpDeny{},
+		clientEgressToEchoDeny{},
+		clientIngressToEchoNamedPortDeny{},
+		clientEgressToEchoExpressionDeny{},
+		clientWithServiceAccountEgressToEchoDeny{},
+		clientEgressToEchoServiceAccountDeny{},
+		clientEgressToCidrDeny{},
+		clientEgressToCidrDenyDefault{},
+		health{},
+		northSouthLoadbalancing{},
+		podToPodEncryption{},
+		nodeToNodeEncryption{},
+		egressGateway{},
+		egressGatewayExcludedCidrs{},
+		podToNodeCidrpolicy{},
+		northSouthLoadbalancingWithL7Policy{},
+		echoIngressL7{},
+		echoIngressL7NamedPort{},
+		clientEgressL7Method{},
+		clientEgressL7{},
+		clientEgressL7NamedPort{},
+		clientEgressL7TlsDenyWithoutHeaders{},
+		clientEgressL7TlsHeaders{},
+		clientEgressL7SetHeader{},
+		echoIngressAuthAlwaysFail{},
+		echoIngressMutualAuthSpiffe{},
+		podToIngressService{},
+		podToIngressServiceDenyAll{},
+		podToIngressServiceDenyIngressIdentity{},
+		podToIngressServiceDenyBackendService{},
+		podToIngressServiceAllowIngressIdentity{},
+		outsideToIngressService{},
+		outsideToIngressServiceDenyWorldIdentity{},
+		outsideToIngressServiceDenyCidr{},
+		outsideToIngressServiceDenyAllIngress{},
+		dnsOnly{},
+		toFqdns{},
+		podToControlplaneHost{},
+		podToK8sOnControlplane{},
+		podToControlplaneHostCidr{},
+		podToK8sOnControlplaneCidr{},
+	}
+)
+
+type factory interface {
+	name() string
```

Comment:
this method seems unnecessary if it's only called from build() 💭

Reply:
Removed.